crates/router/src/core/webhooks/utils.rs

```diff
@@ -154,3 +158,98 @@ pub(crate) fn get_idempotent_event_id(
 pub(crate) fn generate_event_id() -> String {
     common_utils::generate_time_ordered_id("evt")
 }
+
+pub async fn perform_redis_lock<A>(
+    state: &A,
+    unique_locking_key: String,
+    merchant_id: common_utils::id_type::MerchantId,
+) -> RouterResult<bool>
+where
+    A: SessionStateInfo,
+{
+    let redis_conn = state
+        .store()
+        .get_redis_conn()
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+
+    let redis_locking_key = format!(
+        "{}_{}_{}",
+        WEBHOOK_LOCK_PREFIX,
+        merchant_id.get_string_repr(),
+        unique_locking_key
+    );
+    let redis_lock_expiry_seconds = state.conf().lock_settings.redis_lock_expiry_seconds;
+
+    let redis_lock_result = redis_conn
+        .set_key_if_not_exists_with_expiry(
+            &redis_locking_key.as_str().into(),
+            state.get_request_id(),
```

Comment:
Why do we need the request ID here? And what would happen if the request ID is `None`, considering `get_request_id()` returns an `Option<String>`?

Reply:
@SanchithHegde the `get_request_id()` is used as redis value in api_locking, so I had followed the same. But if request_id is None, it could lead to unintended behavior in the redis locking mechanism.

I have changed it to `uuid` now, shouldn't we also do the same change for the api_locking usecases?